.gitlab-ci.yml

```diff
@@ -31,53 +28,78 @@ pull:
     - git push secondary --tags
     - git push secondary --all
 
-build-for-pypi:
+build-pypi:
   rules:
-    - if: '$CI_PIPELINE_SOURCE == "push"'  #run this job only on push event
+    - if: '$CI_PIPELINE_SOURCE == "push"'
+    - if: '$CI_PIPELINE_SOURCE == "schedule"'
   tags:
-    - dinvm
+    - lxd
   stage: build
   script:
     - make CI
   artifacts:
     paths:
       - dist/*
 
-delete-vm:
+build-msi:
   rules:
-    - if: '$CI_PIPELINE_SOURCE == "push"'  #run this job only on push event
+    - if: '$CI_PIPELINE_SOURCE == "push"'
+    - if: '$CI_PIPELINE_SOURCE == "schedule"'
+    - if: '$CI_PIPELINE_SOURCE == "web"'
   tags:
-    - shell
-  stage: cleanup
+    - lxd
+  stage: build
   script:
-    - vboxmanage unregistervm ubuntu-runner-runner-hszxezQU-concurrent-0 --delete
+    - make wine-build
+  artifacts:
+    paths:
+      - wine-build/*.exe
+      - wine-build/*.msi
 
-publish-pypi:
-  when: manual   #run this job only on manual user interaction in the webinterface
-  allow_failure: false
-  only:
-    - pushes
+upload-nightly:
+  stage: deploy
+  rules:
+    - if: '$CI_PIPELINE_SOURCE == "schedule"'
+  tags:
+    - docker
   image: ubuntu:20.04
+  before_script:
+    - apt-get update -y
+    - apt-get install -y git zip
+  script:
+    - chmod +x ./release.sh/nightly_upload.sh
+    - mkdir nightly-build
+    - mv wine-build/*.msi nightly-build
+    - mv wine-build/*.exe nightly-build
+    - zip -r pynitrokey.zip nightly-build
+    - ./release.sh/nightly_upload.sh pynitrokey.zip
+
+publish-pypi:
+  rules:
+    - if: '$CI_PIPELINE_SOURCE == "web"'
   tags:
-    - dinvm
+    - lxd
   stage: deploy
   script:
-    - make update-venv
+    - make CI
     - echo "[distutils]" >> ~/.pypirc
     - echo "index-servers =" >> ~/.pypirc
-    - echo "    test-pypi" >> ~/.pypirc
+    - echo "    pypi" >> ~/.pypirc
     - echo "" >> ~/.pypirc
-    - echo "[test-pypi]" >> ~/.pypirc
+    - echo "[pypi]" >> ~/.pypirc
     - echo "repository = https://upload.pypi.org/legacy/" >> ~/.pypirc
     - echo "username = __token__" >> ~/.pypirc
-    - echo "${PYPI_TOKEN}" >> ~/.pypirc
+    - echo "password=${PYPI_TOKEN}" >> ~/.pypirc
```

Comment:
Missed that - I think in general it's better to use template files (with string replacement e.g. via `sed`) than creating them directly from the script.
Additionally, the script in the `yml` file should be as small as possible, calling other scripts, instead of containing more complex logic.